community/kernel/src/main/java/org/neo4j/kernel/impl/index/labelscan/NativeLabelScanStore.java

```diff
@@ -0,0 +1,265 @@
+            return new NativeLabelScanWriter( inserter, rangeSize, batching ? 10_000 : 1_000 );
+        }
+        catch ( IOException e )
+        {
+            throw new RuntimeException( e );
+        }
+    }
+
+    /**
+     * Forces all changes to {@link PageCache} and creates a checkpoint so that the {@link LabelScanStore}
+     * is recoverable from this point, given that the same transactions which will be applied after this point
+     * and non-clean shutdown will be applied again on next startup.
+     *
+     * @param limiter {@link IOLimiter}.
+     * @throws UnderlyingStorageException on failure writing changes to {@link PageCache}.
+     */
+    @Override
+    public void force( IOLimiter limiter ) throws UnderlyingStorageException
+    {
+        try
+        {
+            index.checkpoint( limiter );
+        }
+        catch ( IOException e )
+        {
+            throw new UnderlyingStorageException( e );
+        }
+    }
+
+    /**
+     * Unsupported by this implementation.
+     *
+     * @return nothing since {@link UnsupportedOperationException} will be thrown.
+     * @throws UnsupportedOperationException since not supported by this implementation.
+     */
+    @Override
+    public AllEntriesLabelScanReader allNodeLabelRanges()
+    {
+        throw new UnsupportedOperationException( "Not supported by " + getClass() );
+    }
+
+    /**
+     * @return store files, namely the single "neostore.labelscanstore.db" store file.
+     * @throws IOException on file access exceptions.
+     */
+    @Override
+    public ResourceIterator<File> snapshotStoreFiles() throws IOException
+    {
+        return asResourceIterator( iterator( storeFile ) );
+    }
+
+    /**
+     * Instantiates the underlying {@link GBPTree} and its resources.
+     *
+     * @throws IOException on {@link PageCache} exceptions.
+     */
+    @Override
+    public void init() throws IOException
+    {
+        index = new GBPTree<>( pageCache, storeFile, new LabelScanLayout( rangeSize ), pageSize, NO_MONITOR );
+    }
+
+    /**
+     * Starts the store and makes it available for queries and updates.
+     * Any required recovery must take place before calling this method.
+     *
+     * @throws IOException on {@link PageCache} exceptions.
+     */
+    @Override
+    public void start() throws IOException
+    {
+        if ( isEmpty() )
+        {
+            LabelScanStoreProvider.rebuild( this, fullStoreChangeStream );
+        }
+        started = true;
+    }
+
+    private boolean isEmpty() throws IOException
+    {
+        try ( RawCursor<Hit<LabelScanKey,LabelScanValue>,IOException> cursor = index.seek(
+                new LabelScanKey().set( 0, 0 ),
+                new LabelScanKey().set( Integer.MAX_VALUE, Long.MAX_VALUE ) ) )
+        {
+            return !cursor.next();
+        }
+    }
+
+    @Override
+    public void stop() throws IOException
+    {   // Not needed
```

Comment:
Maybe, but `started==false` doesn't prevent new writers and LuceneLabelScanStore doesn't do anything in `stop()` either